credentials/xds/xds_client_test.go

```diff
@@ -181,6 +210,30 @@ func testServerMutualTLSHandshake(rawConn net.Conn) handshakeResult {
 	return handshakeResult{connState: conn.ConnectionState()}
 }
 
+// A handshake function which simulates a successful handshake with mutual
+// authentication.
+func testServerMutualTLSHandshakeSPIFFE(rawConn net.Conn) handshakeResult {
+	cert, err := tls.LoadX509KeyPair(testdata.Path("spiffe_end2end/client_spiffe.pem"), testdata.Path("spiffe_end2end/client.key"))
+	if err != nil {
+		return handshakeResult{err: err}
+	}
+	pemData, err := os.ReadFile(testdata.Path("x509/client_ca_cert.pem"))
+	if err != nil {
+		return handshakeResult{err: err}
+	}
+	roots := x509.NewCertPool()
+	roots.AppendCertsFromPEM(pemData)
+	cfg := &tls.Config{
```

Comment:
To confirm, this is the server-side TLS config - correct? If yes, shouldn't we be setting request and require client cert?